src/clj/fluree/db/reasoner/owl_datalog.cljc

```diff
@@ -0,0 +1,850 @@
+                     get-all-ids)
+        property (get-id owl-statement)
+        rule     {"where"  {"@id"    nil,
+                            property "?ps"}
+                  "insert" {"@id"   "?ps"
+                            "@type" range}}]
+    ;; rule-id *is* the property
+    (if (and property (seq range))
+      (conj all-rules [(str property "(rdfs:range)") rule])
+      (do (log/warn "Ignoring rdfs:range rule " owl-statement
+                    " as property or range definition not supported by owl2rl reasoning rule.")
+          all-rules))))
+
+;; owl:FunctionalProperty
+(defmethod to-datalog ::prp-fp
+  [_ _ owl-statement all-rules]
+  (let [fp   (get-id owl-statement)
+        rule {"where"  [{"@id" "?s"
+                         fp    "?fp-vals"}
+                        {"@id" "?s"
+                         fp    "?fp-vals2"}
+                        ["filter" "(not= ?fp-vals ?fp-vals2)"]]
+              "insert" {"@id"                "?fp-vals"
+                        const/iri-owl:sameAs "?fp-vals2"}}]
+    (conj all-rules [(str const/iri-owl:FunctionalProperty "(" fp ")") rule])))
+
+;; owl:InverseFunctionalProperty
+(defmethod to-datalog ::prp-ifp
+  [_ _ owl-statement all-rules]
+  (let [ifp  (get-id owl-statement)
+        rule {"where"  [{"@id" "?x1"
+                         ifp   "?y"}
+                        {"@id" "?x2"
+                         ifp   "?y"}
+                        ["filter" "(not= ?x1 ?x2)"]]
+              "insert" {"@id"                "?x1"
+                        const/iri-owl:sameAs "?x2"}}]
+    (conj all-rules [(str const/iri-owl:InverseFunctionalProperty "(" ifp ")") rule])))
+
+;;owl:SymetricProperty
+(defmethod to-datalog ::prp-symp
+  [_ _ owl-statement all-rules]
+  (let [symp (get-id owl-statement)
+        rule {"where"  [{"@id" "?x"
+                         symp  "?y"}]
+              "insert" {"@id" "?y"
+                        symp  "?x"}}]
+    (conj all-rules [(str symp "(owl:SymetricProperty)") rule])))
+
+;; owl:TransitiveProperty
+(defmethod to-datalog ::prp-trp
+  [_ _ owl-statement all-rules]
+  (let [trp  (get-id owl-statement)
+        rule {"where"  [{"@id" "?x"
+                         trp   "?y"}
+                        {"@id" "?y"
+                         trp   "?z"}]
+              "insert" {"@id" "?x"
+                        trp   "?z"}}]
+    (conj all-rules [(str trp "(owl:TransitiveProperty)") rule])))
+
+;; rdfs:subPropertyOf
+(defmethod to-datalog ::prp-spo1
+  [_ inserts owl-statement all-rules]
+  (let [child-prop   (get-id owl-statement)
+        parent-props (-> owl-statement
+                         (get const/iri-rdfs:subPropertyOf)
+                         get-all-ids)
+        triples      (mapv (fn [parent-prop]
+                             [child-prop const/iri-rdfs:subPropertyOf {"@id" parent-prop}])
+                           parent-props)
+        rule-id      (str child-prop "(rdfs:subPropertyOf)")]
```

Comment:
Could we run into any issues using the compact form here? Could users ever see these rule-ids, and could they be affected by iris and/or contexts they set?

Reply:
No issues, the rule IDs must just be unique strings that are informative enough to understand where they originated. They are IRIs, but technically don't need to be.

I settled in on: <source-IRI>(<rule-type>)#hash - where hash is only used in a few circumstances.

As I got more into this, I saw value in not just a string storing the source of the rule, but a map that can hold more data. I think if I started over I might favor that approach, but we'd still need a unique rule identifier.